rclcpp_lifecycle/test/test_state_machine_info.cpp

```diff
@@ -54,8 +54,30 @@ TEST_F(TestStateMachineInfo, available_states) {
 
 TEST_F(TestStateMachineInfo, available_transitions) {
   auto test_node = std::make_shared<rclcpp_lifecycle::LifecycleNode>("testnode");
-  std::vector<rclcpp_lifecycle::Transition> available_transitions =
+  std::vector<rclcpp_lifecycle::Transition> current_available_transitions =
     test_node->get_available_transitions();
+  EXPECT_EQ(2u, current_available_transitions.size());
```

Comment:
I think a better test of `get_available_transitions` would be to check the available transitions from a couple of different states and ensure they match the expected transitions by id

Reply:
okay, sgtm. i would add more concrete test cases for `get_available_transitions `.

Reply:
more test scenarios added along with each primary state.